### ISSUE TITLE
Train model by myself

### ISSUE THREAD

**Issue by DenisFromRussia** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#2]
Thanks for the code, but is there a possibility to train the model before self-adaptation without pretrained weights, if yes, what do i need to do?

**Comment by AlessioTonioni** [comment]
Hi,
Unfortunately, we do not provide the code to train the model from scratch since we have it available on a more complex private framework. 
Anyway you should be able to modify easily [Stereo_Online_Adaptation.py](https://github.com/CVLAB-Unibo/Real-time-self-adaptive-deep-stereo/blob/master/Stereo_Online_Adaptation.py) for supervised training from scratch.
0) Remove the assert on line 151 to allow the model to start from random initialization.
1) Modify the dataloader to handle multiple epochs, bigger batch size and set the proper flag. It should already handle the loading of PFM gt files, so it can be used also for FlyingThings3D.
 E.g. :
```python
data_set = data_reader.dataset(
			args.list,
			batch_size = 4,
			crop_shape=args.imageShape,
			num_epochs=40,
			augment=True,
			is_training=True,
			shuffle=True
)
```
2) Modify the loss function being optimized to be supervised L1
```python
full_reconstruction_loss = loss_factory.get_supervised_loss('sum_l1',multiscale=True,reduced=True)(predictions,inputs)
```
3) Implement the training loop. You can also take inspiration from [this old repository of mine](https://github.com/AlessioTonioni/dispflownet-tf)

At some point in the near future I might add a stand alone script to train the model from scratch, but unfrtunatelly I have not write it yet.

At some point
